compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/ir/IrUtils.kt

```diff
@@ -37,3 +41,17 @@ val IrType.erasedUpperBound: IrClass
         is IrTypeParameterSymbol -> classifier.owner.erasedUpperBound
         else -> throw IllegalStateException()
     }
+
+val IrFunction.propertyIfAccessor: IrDeclaration
+    get() = (this as? IrSimpleFunction)?.correspondingPropertySymbol?.owner ?: this
+
+fun IrFunction.annotatedWithJvmDefault(): Boolean = propertyIfAccessor.hasAnnotation(JVM_DEFAULT_FQ_NAME)
+
+fun IrFunction.isJvmDefault(context: JvmBackendContext): Boolean = annotatedWithJvmDefault() &&
+        context.state.target.bytecodeVersion >= JvmTarget.JVM_1_8.bytecodeVersion &&
```

Comment:
Version check is performed in frontend and here it's redundant: IrFunction could be from another module compiled to 1.8 and this one is compiling to 1.6

Reply:
jvmDefaultMode.isEnabled check is also performed in frontend

Reply:
Thanks for pointing that out. I've updated it accordingly.